src/contract/assembled_transaction.ts

```diff
@@ -682,7 +685,17 @@ export class AssembledTransaction<T> {
       );
     }
 
-    if (!signTransaction) {
+    // Check if signTransaction is a Keypair
+    let signFunction: SignTransaction | undefined;
+
+    if (signTransaction instanceof Keypair) {
+      const keypair = signTransaction;
+      signFunction = basicNodeSigner(keypair, this.options.networkPassphrase).signTransaction;
+    } else if (typeof signTransaction === 'function') {
+      signFunction = signTransaction;
+    }
+
+    if (!signFunction) {
```

Comment:
This could be less verbose, and arguably safer...
`signFunction` doesn't need to be a let it can be a const

```suggestion
if (!signTransaction && !keyPair) {
      throw new AssembledTransaction.Errors.NoSigner(
        "You must provide a signTransaction function, or a keypair either when calling " +
        "`signAndSend` or when initializing your Client"
      );
    };
    
    const signFunction = signTransaction ? signTransaction : basicNodeSigner(keypair, this.options.networkPassphrase).signTransaction
```

However you'll also need to remove lines 699-703 also.

Further, you shouldn't modify the type SignTransaction you should just add keypair to clientoptions as an optional and update the `sign` function here around line 664... you also need to look at the `signandsend` function for consistency and update that correctly.